app/views/concerns/facet_presenter.rb

```diff
@@ -125,4 +127,39 @@ def range_facet(facet)
       date_end: range_max
     }
   end
+
+  def boolean_facet(facet)
+    {
+      url: boolean_facet_url(facet),
+      text: facet_map(facet.name),
+      is_checked: boolean_facet_checked?(facet)
+    }
+  end
+
+  def boolean_facet_url(facet)
+    facet_config = blacklight_config.facet_fields[facet.name]
+
+    if boolean_facet_in_params?(facet.name)
+      search_action_url(remove_facet_params(facet.name, facet_params(facet.name).first, params))
+    elsif boolean_facet_checked?(facet)
+      search_action_url(add_facet_params_and_redirect(facet.name, facet_config.boolean[:off]))
+    else
+      search_action_url(add_facet_params_and_redirect(facet.name, facet_config.boolean[:on]))
+    end
+  end
+
+  def boolean_facet_checked?(facet)
```

Comment:
Assignment Branch Condition size for boolean_facet_checked? is too high. [21.4/15]